akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -100,13 +60,21 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
   override def behavior: Behavior[Command] = Behaviors.setup[Any] { ctx ⇒
 
     val setup = new Setup(ctx)
+    val registry = SuperServiceRegistry(setup.settings.distributedKeyCount)
 
     // subscribe to changes from other nodes
     val replicatorMessageAdapter: ActorRef[Replicator.ReplicatorMessage] =
       ctx.messageAdapter[Replicator.ReplicatorMessage] {
-        case changed @ Replicator.Changed(ReceptionistKey) ⇒ ChangeFromReplicator(changed.get(ReceptionistKey))
+        case changed: Replicator.Changed[_] @unchecked ⇒
+          // FIXME couldn't think of a cleaner way to do this ;(
```

Comment:
I think this is fine, since the keys are dynamic. One way might be to instead loop the `superRegistry.allDdataKeys` until matching the key in `Changed` and then grab the value with `changed.get`. Perhaps prettier but not as efficient.